internal/controller/clusterdeployment_controller.go

```diff
@@ -215,109 +210,104 @@ func (r *ClusterDeploymentReconciler) reconcileUpdate(ctx context.Context, cd *k
 }
 
 func (r *ClusterDeploymentReconciler) updateCluster(ctx context.Context, cd *kcm.ClusterDeployment, clusterTpl *kcm.ClusterTemplate) (ctrl.Result, error) {
-	l := ctrl.LoggerFrom(ctx)
-
 	if clusterTpl == nil {
 		return ctrl.Result{}, errors.New("cluster template cannot be nil")
 	}
 
+	l := ctrl.LoggerFrom(ctx)
+
+	r.initClusterConditions(cd)
+
 	if !clusterTpl.Status.Valid {
-		errMsg := "provided template is not marked as valid"
-		apimeta.SetStatusCondition(cd.GetConditions(), metav1.Condition{
-			Type:    kcm.TemplateReadyCondition,
-			Status:  metav1.ConditionFalse,
-			Reason:  kcm.FailedReason,
-			Message: errMsg,
-		})
-		return ctrl.Result{}, errors.New(errMsg)
+		err := fmt.Errorf("ClusterTemplate %s is not marked as valid", client.ObjectKeyFromObject(clusterTpl))
+		r.setCondition(cd, kcm.TemplateReadyCondition, err)
+		if r.IsDisabledValidationWH {
+			l.Error(err, "template is not valid, will not retrigger this error")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
 	}
+	r.setCondition(cd, kcm.TemplateReadyCondition, nil)
 	// template is ok, propagate data from it
 	cd.Status.KubernetesVersion = clusterTpl.Status.KubernetesVersion
 
-	apimeta.SetStatusCondition(cd.GetConditions(), metav1.Condition{
-		Type:    kcm.TemplateReadyCondition,
-		Status:  metav1.ConditionTrue,
-		Reason:  kcm.SucceededReason,
-		Message: "Template is valid",
-	})
+	var cred *kcm.Credential
```

Comment:
nit: Do we really need to initialize `cred` var here? It's used at line 255, and then at line 297 it's reassigned

Reply:
this is done to avoid double-fetch, if the webhook is disabled a `Credential` object is being fetch and assigned to the variable during the validation and then reused to extract identity reference